### DISCUSSION TITLE
could EventReader add fn 'is_not_empty_clear()' ?

### DISCUSSION THREAD

**Discussion by alexniver** [bevyengine/bevy#5840] [Ideas]
```
        if !events.is_empty {
		events.clear();
		// Process the fact that one or more event was received
	}
```

this code used a lot, and new bevy coder always forget add ```events.clear()```.

maybe ```is_not_empty_clear``` is needed?

**Comment by DJMcNab** [comment]
The previous discussion of this question happened in https://github.com/bevyengine/bevy/pull/4693

I believe that even if such a method is added, the pit of failure of using 'normal' `is_empty` would still be there - if users haven't read the `is_empty` docs, they wouldn't know they need to use `is_not_empty_clear` , just as they don't know they need to use `clear` now.

I think the most feasible fix is the one I suggested in https://github.com/bevyengine/bevy/pull/4693#issuecomment-1121099893. Basically, we should recognise that leaving an uncleared `EventReader` is always a logic error, so warn or panic if that occurs.

**Reply by alexniver** [reply]
oh, I like the panic idea. at least know something went wrong.

**Reply by tim-blackbird** [reply]
This would be nice as a debug assertion :+1:
